csrc/device_lower/pass/allocation.cpp

```diff
@@ -819,10 +819,8 @@ class AllocationInserter : public kir::ExprMutator {
     lower_alloc_info_map[alloc_expr] = std::move(lower_alloc_info_ptr);
   }
 
-  void handle(kir::IfThenElse*) final {
-    NVF_THROW(
```

Comment:
Why is this necessary?

Reply:
Because we are now inserting a
```C++
if (first warp) {
  tmem::alloc
}
```
in the allocation pass

Reply:
I understand that's the reason for the same changes of the other passes, but here, this is the pass inserting the `IfThenElse`-based allocation op, so I suppose this `handle` should never be called yet. What am I missing?